### ISSUE TITLE
Do you need anything from xterm.js?

### ISSUE THREAD

**Issue by Tyriar** [Eugeny/terminus#542]
Hi @Eugeny, I'm one of the maintainers on xterm.js. I noticed that hterm appears to be the default frontend and was wondering if there were particular issues on our side for this?

**Comment by MrKumaPants** [comment]
@Tyriar How about getting to work on TrueColor? :/

**Comment by Tyriar** [comment]
@MrKumaPants we're getting fairly close to done with the buffer improvements that will allow us to do true color without taking way too much memory https://github.com/xtermjs/xterm.js/issues/791

**Comment by Eugeny** [comment]
Hey Daniel, thanks for reaching out!

I've recently posted two feature requests that would definitely make it easier to switch over to xterm in the future:

https://github.com/xtermjs/xterm.js/issues/1803
https://github.com/xtermjs/xterm.js/issues/1824

I'm definitely looking forward to the WebGL renderer as well.

One major pain of integrating xterm is how monkeypatching-resistant it is - with a lot of stuff locked away within closures - e.g. it's absolutely impossible to patch any event handlers installed with `addDisposableDomListener()`, and the function itself is unreachable from `xterm` when it's built. 

I've considered importing `xterm/src/xterm.js` directly, which would make all submodules importable on our side as well, but that doesn't work nicely since Terminus has stricter code policy (`noUnusedLocals` etc), and it results in a ton of build warnings, which TypeScript can't ignore on per-module basis.

**Comment by Eugeny** [comment]
Also, what's the state of ConPTY integration in node-pty? The PR looks feature-complete, but I hadn't have a chance to test it yet. I wouldn't mind adding it to Terminus before it's released/stable.

**Comment by Tyriar** [comment]
I commented on those 2 issues.

> One major pain of integrating xterm is how monkeypatching-resistant it is - with a lot of stuff locked away within closures - e.g. it's absolutely impossible to patch any event handlers installed with addDisposableDomListener(), and the function itself is unreachable from xterm when it's built.

Do you have some examples of what you're trying to do here?

> I'm definitely looking forward to the WebGL renderer as well.

Ditto, it's not a priority to push that in currently but I'm going to keep the PR up to date in the meantime.

> Also, what's the state of ConPTY integration in node-pty? The PR looks feature-complete, but I hadn't have a chance to test it yet. I wouldn't mind adding it to Terminus before it's released/stable.

It doesn't work in Electron 2 so it's been blocked on VS Code adopting Electron 3, which is likely to happen in the next release so expect me to clean up and merge that PR soon 😃

**Comment by Eugeny** [comment]
Thanks! 
As to monkey-patching, one example would be hooking the paste events.

**Comment by Eugeny** [comment]
@MrKumaPants I've just added True Color support in #23

**Comment by Eugeny** [comment]
xterm frontend is now the default
